test/src/tests-batchUploader.ts

```diff
@@ -1716,5 +1717,110 @@ describe('batch uploader', () => {
         .catch((e) => {
         })
         });
+
+        describe('noFunctional', () => {
+            const eventStorageKey = 'mprtcl-v4_abcdef-events';
+            const batchStorageKey = 'mprtcl-v4_abcdef-batches';
+            beforeEach(() => {
+                window.mParticle._resetForTests(MPConfig);
+            });
+
+            it('should store events in session storage when noFunctional is default (false)', async () => {
+                window.mParticle.config.flags = {
+                    offlineStorage: '100',
+                    ...enableBatchingConfigFlags,
+                };
+                window.mParticle.init(apiKey, window.mParticle.config);
+                await waitForCondition(hasIdentifyReturned);
+                const mpInstance = window.mParticle.getInstance();
+                const uploader = mpInstance._APIClient.uploader;
+                uploader.queueEvent(event0);
+                expect(!!window.sessionStorage.getItem(eventStorageKey)).to.equal(true);
+            });
+
+            it('should NOT store events in session storage when noFunctional is true', async () => {
+                window.mParticle.config.flags = {
+                    offlineStorage: '100',
+                    ...enableBatchingConfigFlags,
+                };
+                window.mParticle.config.launcherOptions = {
+                    ...(window.mParticle.config.launcherOptions || {}),
+                    noFunctional: true,
+                };
```

Comment:
is there a reason to spread what was previously existing on launcherOptions?  i'd expect just setting the noFunctional:true or false in each of these would be more appropriate so that we know exactly what is going in the `launcherOptions` for each test.  same for all the tests below

```suggestion
                window.mParticle.config.launcherOptions = {
                    noFunctional: true,
                };
```